--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,14 +70,15 @@
 #------------------------------------------------------------------------------
 # Searching for CUDA
 #------------------------------------------------------------------------------
-if(NOT APPLE)
+option(WITH_CUDA "Compile with CUDA support" ON)
+if(NOT APPLE AND WITH_CUDA)
 find_package( CUDA 8)
 if(CUDA_FOUND)
   message(STATUS "Found CUDA")
     include_directories(${CUDA_INCLUDE_DIRS})
     list(APPEND LVR2_DEFINITIONS -DLVR2_USE_CUDA)
 endif(CUDA_FOUND)
-endif(NOT APPLE)
+endif(NOT APPLE AND WITH_CUDA)

 #------------------------------------------------------------------------------
 # Searching for OpenCV
